pkg/cloudprovider/aws/aws.go

```diff
@@ -109,73 +112,145 @@ type ec2InstanceFilter struct {
 }
 
 // True if the passed instance matches the filter
-func (f *ec2InstanceFilter) Matches(instance ec2.Instance) bool {
-	if f.PrivateDNSName != "" && instance.PrivateDNSName != f.PrivateDNSName {
+func (f *ec2InstanceFilter) Matches(instance *ec2.Instance) bool {
+	if f.PrivateDNSName != "" && *instance.PrivateDNSName != f.PrivateDNSName {
 		return false
 	}
 	return true
 }
 
-// goamzEC2 is an implementation of the EC2 interface, backed by goamz
-type goamzEC2 struct {
+// awsSdkEC2 is an implementation of the EC2 interface, backed by aws-sdk-go
+type awsSdkEC2 struct {
 	ec2 *ec2.EC2
 }
 
+func stringPointerArray(orig []string) []*string {
+	n := make([]*string, len(orig))
+	for i, s := range orig {
+		n[i] = &s
```

Comment:
I think you want
```
       for i := range orig {
               n[i] = &orig[i]
```

because otherwise you are taking the address of the same string variable every time because of a go-oddity:

http://play.golang.org/p/ruMYQ1uxAk